src/fidesops/api/v1/endpoints/dataset_endpoints.py

```diff
@@ -166,35 +176,95 @@ def patch_datasets(
             "fides_key": dataset.fides_key,
             "dataset": dataset.dict(),
         }
-        try:
-            if connection_config.connection_type == ConnectionType.saas:
-                _validate_saas_dataset(connection_config, dataset)
-            # Try to find an existing DatasetConfig matching the given connection & key
-            dataset_config = DatasetConfig.create_or_update(db, data=data)
-            created_or_updated.append(dataset_config.dataset)
-        except (SaaSConfigNotFoundException, ValidationError) as exception:
-            logger.warning(exception.message)
-            failed.append(
-                BulkUpdateFailed(
-                    message=exception.message,
-                    data=data,
-                )
-            )
-        except Exception:
-            logger.warning(f"Create/update failed for dataset '{data['fides_key']}'.")
-            failed.append(
-                BulkUpdateFailed(
-                    message="Dataset create/update failed.",
-                    data=data,
-                )
-            )
+        create_or_update_dataset(
+            connection_config, created_or_updated, data, dataset, db, failed
+        )
+    return BulkPutDataset(
+        succeeded=created_or_updated,
+        failed=failed,
+    )
+
 
+@router.patch(
+    YAML_DATASETS,
+    dependencies=[Security(verify_oauth_client, scopes=[DATASET_CREATE_OR_UPDATE])],
+    status_code=200,
+    response_model=BulkPutDataset,
+    include_in_schema=False  # Not including this path in the schema.
+    # Since this yaml function needs to access the request, the open api spec will not be generated correctly.
+    # To include this path, extend open api: https://fastapi.tiangolo.com/advanced/extending-openapi/
```

Comment:
This is fine for now but we should find a way to expose this information in our Swagger API UI if we add additional YAML endpoints. Can we document this endpoint somewhere if we haven't already?

Reply:
I brought this up with @seanpreston & @conceptualshark. We can have a subsequent ticket. 

I would think our solution would depend on how many resources and actions we will support in the yaml API. For me there are open questions: 

- How much will the JSON and YAML APIs diverge? 
- Is it going to be a large API? If so we might want to find a tool to autogenerate open api spec. 
- Do we want a separate spec for the yaml api?